app/ota_client.py

```diff
@@ -556,10 +569,12 @@ def __init__(
         self,
         store: OtaClientStatistics,
         *,
+        max_concurrency_tasks: int,
         interval=1,
     ) -> None:
         self.abort_event = threading.Event()
         self.last_error = None
+        self.se = threading.Semaphore(max_concurrency_tasks)
```

Comment:
In the previous review, semaphore member is removed from this class, and semaphore is used as callback argument.
Is this change intended?

Reply:
Yes, it is intended.
Previously this semaphore is created outside `_CreateRegularStatsCollector`, and used as a parameter for callback, and directly used in the threadpool for loop. 
Now I change to initialize the semaphore in the `_CreateRegularStatsCollector` class, and the this semaphore is accessed as `_CreateRegularStatsCollector` instance's property in the for loop.

What I want to achieve is to let the Collector instance take the whole control of acquire/release semaphore. (Oh for better understanding, maybe I should wrap `_collector.se.acquire()` in line 1107 into a method?)

Reply:
Ah I may also let the collector control the exception detection.
https://github.com/tier4/ota-client/blob/00ca4b70bb009ee2a67817ac7c4da12626676c59/app/ota_client.py#L1119-L1132

Reply:
> should wrap _collector.se.acquire() in line 1107 into a method

Yes!, I think it is better.